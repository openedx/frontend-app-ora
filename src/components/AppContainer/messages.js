import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  ERR_INVALID_STATE_FOR_ASSESSMENT: {
    defaultMessage: 'This step is not available. Unable to retrieve the assessment.',
    description: 'Error message for invalid state for assessment',
    id: 'frontend-app-ora.StatusAlerts.ERR_INVALID_STATE_FOR_ASSESSMENT',
  },
  unknownError: {
    defaultMessage: 'An unknown error occurred. Please try again.',
    description: 'Error message for unknown error',
    id: 'frontend-app-ora.StatusAlerts.unknownError',
  },
  errorHeader: {
    defaultMessage: 'Something went wrong.',
    description: 'Error message header',
    id: 'frontend-app-ora.StatusAlerts.errorHeader',
  },
});

export default messages;
