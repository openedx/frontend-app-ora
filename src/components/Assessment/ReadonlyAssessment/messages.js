import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  readMore: {
    id: 'frontend-app-ora.readMore',
    defaultMessage: 'Read more',
    description: 'Read more button text',
  },
  readLess: {
    id: 'frontend-app-ora.readLess',
    defaultMessage: 'Read less',
    description: 'Read less button text',
  },
  unweightedGrade: {
    id: 'ora-collapsible-comment.unweightedGrade',
    defaultMessage: '{stepLabel} grade',
    description: 'Unweighted grade group text',
  },
  grade: {
    id: 'frontend-app-ora.grade',
    defaultMessage: '{stepLabel} grade:',
    description: 'Grade button text',
  },
  gradePoints: {
    id: 'frontend-app-ora.gradePoints',
    defaultMessage: '{earned} / {possible}',
    description: 'Grade points button text',
  },
  notWeightedGradeLabel: {
    id: 'frontend-app-ora.notWeightedGradeLabel',
    defaultMessage: '(Not weighted toward final grade))',
    description: 'Not weighted grade label',
  },
  overallFeedback: {
    id: 'frontend-app-ora.overallFeedback',
    defaultMessage: 'Overall feedback',
    description: 'Overall feedback label',
  },
  stepComments: {
    id: 'ora-collapsible-comment.stepComments',
    defaultMessage: '{step} comments',
    description: 'Step comments label for completed assessments',
  },
  comments: {
    id: 'ora-collapsible-comment.comments',
    defaultMessage: 'comments',
    description: 'Comments label for completed assessments',
  },
  submittedGrade: {
    id: 'ora-collapsible-comment.submittedGrade',
    defaultMessage: 'Submitted grade',
    description: 'Header for submitted grade display',
  },
  submittedAssessment: {
    id: 'ora-collapsible-comment.submittedAssessment',
    defaultMessage: 'Submitted assessment',
    description: 'Header for submitted assessment display',
  },
  comment: {
    id: 'ora-collapsible-comment.comment',
    defaultMessage: 'Comments',
    description: 'Header for comment component if no step provided',
  },
  stepComment: {
    id: 'ora-collapsible-comment.stepComment',
    defaultMessage: '{step} comment',
    description: 'Header for comment component if step is provided',
  },
  points: {
    id: 'ora-collapsible-comment.points',
    defaultMessage: 'Points',
    description: 'Header for points component',
  },
});

export default messages;
