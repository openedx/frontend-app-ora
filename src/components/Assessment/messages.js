import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  gradeSubmitted: {
    id: 'frontend-app-ora.EditableAssessment.gradeSubmitted',
    defaultMessage: 'Grade Submitted',
    description: 'Submit Grade button text after successful submission',
  },
  rubric: {
    id: 'frontend-app-ora.EditableAssessment.rubric',
    defaultMessage: 'Rubric',
    description: 'Rubric interface label',
  },
  submitGrade: {
    id: 'frontend-app-ora.EditableAssessment.submitGrade',
    defaultMessage: 'Submit grade',
    description: 'Submit Grade button text',
  },
  submittingGrade: {
    id: 'frontend-app-ora.EditableAssessment.submittingGrade',
    defaultMessage: 'Submitting grade',
    description: 'Submit Grade button text while submitting',
  },
  overallComments: {
    id: 'frontend-app-ora.EditableAssessment.overallComments',
    defaultMessage: 'Overall comments',
    description: 'Rubric overall comments label',
  },
  addComments: {
    id: 'frontend-app-ora.EditableAssessment.addComments',
    defaultMessage: 'Add comments (Optional)',
    description: 'Rubric comments input label',
  },
  comments: {
    id: 'frontend-app-ora.EditableAssessment.comments',
    defaultMessage: 'Comments (Optional)',
    description: 'Rubric comments display label',
  },
  overallFeedbackError: {
    id: 'frontend-app-ora.EditableAssessment.overallFeedbackError',
    defaultMessage: 'The overall feedback is required',
    description: 'Error message when feedback input is required',
  },
  finishLater: {
    id: 'frontend-app-ora.EditableAssessment.finishLater',
    defaultMessage: 'Finish grading later',
    description: 'Button text for close modal action in the assessment view',
  },
});

export default messages;
