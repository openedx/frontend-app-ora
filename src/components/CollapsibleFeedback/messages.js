import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  readMore: {
    id: 'ora-collapsible-comment.readMore',
    defaultMessage: 'Read more',
    description: 'Read more button text',
  },
  readLess: {
    id: 'ora-collapsible-comment.readLess',
    defaultMessage: 'Read less',
    description: 'Read less button text',
  },
  grade: {
    id: 'ora-collapsible-comment.grade',
    defaultMessage: '{stepLabel} Grade:',
    description: 'Grade button text',
  },
  gradePoints: {
    id: 'ora-collapsible-comment.gradePoints',
    defaultMessage: '{earned} / {possible}',
    description: 'Grade points button text',
  },
  notWeightedGradeLabel: {
    id: 'ora-collapsible-comment.notWeightedGradeLabel',
    defaultMessage: '(Not weighted toward final grade))',
    description: 'Not weighted grade label',
  },
  overallFeedback: {
    id: 'ora-collapsible-comment.overallFeedback',
    defaultMessage: 'Overall Feedback',
    description: 'Overall feedback label',
  },
});

export default messages;
