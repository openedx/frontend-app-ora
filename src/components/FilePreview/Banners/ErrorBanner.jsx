import React from 'react';
import PropTypes from 'prop-types';

import { Alert, Button } from '@edx/paragon';
import { Info } from '@edx/paragon/icons';
import { useIntl } from '@edx/frontend-platform/i18n';

const messageShape = PropTypes.shape({
  id: PropTypes.string,
  defaultMessage: PropTypes.string,
});

export const ErrorBanner = ({ actions, headerMessage, children }) => {
  const { formatMessage } = useIntl();
  const actionButtons = actions.map(action => (
    <Button key={action.id} onClick={action.onClick} variant="outline-primary">
      {formatMessage(action.message)}
    </Button>
  ));

  return (
    <Alert variant="danger" icon={Info} actions={actionButtons}>
      <Alert.Heading>
        {formatMessage(headerMessage)}
      </Alert.Heading>
      {children}
    </Alert>
  );
};
ErrorBanner.defaultProps = {
  actions: [],
  children: null,
};
ErrorBanner.propTypes = {
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      onClick: PropTypes.func,
      message: messageShape,
    }),
  ),
  headerMessage: messageShape.isRequired,
  children: PropTypes.node,
};

export default ErrorBanner;
