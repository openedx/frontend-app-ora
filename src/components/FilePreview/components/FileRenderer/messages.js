import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  fileInfo: {
    id: 'frontend-app-ora.InfoPopover.fileInfo',
    defaultMessage: 'File info',
    description: 'Popover trigger button text for file preview card',
  },
  retryButton: {
    id: 'frontend-app-ora.ResponseDisplay.FileRenderer.retryButton',
    defaultMessage: 'Retry',
    description: 'Retry button for error in file renderer',
  },
});

export default messages;
