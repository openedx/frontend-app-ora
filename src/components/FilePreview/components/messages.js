import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  fileNotFoundError: {
    id: 'frontend-app-ora.ResponseDisplay.FileRenderer.fileNotFound',
    defaultMessage: 'File not found',
    description: 'File not found error message',
  },
  unknownError: {
    id: 'frontend-app-ora.ResponseDisplay.FileRenderer.unknownError',
    defaultMessage: 'Unknown errors',
    description: 'Unknown errors message',
  },
});

export default messages;
