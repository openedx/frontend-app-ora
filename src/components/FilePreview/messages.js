import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  fileInfo: {
    id: 'ora-grading.InfoPopover.fileInfo',
    defaultMessage: 'File info',
    description: 'Popover trigger button text for file preview card',
  },
  retryButton: {
    id: 'ora-grading.ResponseDisplay.FileRenderer.retryButton',
    defaultMessage: 'Retry',
    description: 'Retry button for error in file renderer',
  },
  fileNotFoundError: {
    id: 'ora-grading.ResponseDisplay.FileRenderer.fileNotFound',
    defaultMessage: 'File not found',
    description: 'File not found error message',
  },
  unknownError: {
    id: 'ora-grading.ResponseDisplay.FileRenderer.unknownError',
    defaultMessage: 'Unknown errors',
    description: 'Unknown errors message',
  },
});

export default messages;
