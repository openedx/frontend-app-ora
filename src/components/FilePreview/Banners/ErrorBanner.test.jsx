import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import ErrorBanner from './ErrorBanner';

import messages from '../messages';

describe('Error Banner component', () => {
  const children = <p>Abitary Child</p>;

  const props = {
    actions: [
      {
        id: 'action1',
        onClick: jest.fn().mockName('action1.onClick'),
        message: messages.retryButton,
      },
      {
        id: 'action2',
        onClick: jest.fn().mockName('action2.onClick'),
        message: messages.retryButton,
      },
    ],
    headerMessage: messages.unknownError,
    children,
  };

  let el;
  beforeEach(() => {
    el = shallow(<ErrorBanner {...props} />);
  });

  test('snapshot', () => {
    expect(el.snapshot).toMatchSnapshot();
  });

  describe('component', () => {
    test('children node', () => {
      el.instance.findByType('p');
      const childEl = shallow(children);
      el.instance.findByType('p')[0].matches(childEl);
    });

    test('verify actions', () => {
      const { actions } = el.instance.findByType('Alert')[0].props;
      expect(actions).toHaveLength(props.actions.length);

      actions.forEach((action, index) => {
        expect(action.type).toEqual('Button');
        expect(action.props.onClick).toEqual(props.actions[index].onClick);
        // action message
        expect(action.props.children).toEqual(props.actions[index].message.defaultMessage);
      });
    });
  });
});
