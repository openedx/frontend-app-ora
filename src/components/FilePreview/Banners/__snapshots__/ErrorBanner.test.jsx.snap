// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Banner component snapshot 1`] = `
<Alert
  actions={
    Array [
      <Button
        onClick={[MockFunction action1.onClick]}
        variant="outline-primary"
      >
        Retry
      </Button>,
      <Button
        onClick={[MockFunction action2.onClick]}
        variant="outline-primary"
      >
        Retry
      </Button>,
    ]
  }
  icon={[Function]}
  variant="danger"
>
  <Alert.Heading>
    Unknown errors
  </Alert.Heading>
  <p>
    Abitary Child
  </p>
</Alert>
`;
